src/query/functions-v2/tests/it/scalars/testdata/boolean.txt

```diff
@@ -7,15 +7,51 @@ output domain  : {FALSE}
 output         : false
 
 
-ast            : null AND false
-raw expr       : and(NULL, false)
-checked expr   : and<NULL, Boolean NULL>(NULL, CAST(false AS Boolean NULL))
+ast            : true AND null
+raw expr       : and(true, NULL)
+checked expr   : and<Boolean NULL, Boolean NULL>(CAST(true AS Boolean NULL), CAST(NULL AS Boolean NULL))
+optimized expr : false
+output type    : Boolean NULL
+output domain  : Unknown
+output         : false
+
+
+ast            : true AND true
+raw expr       : and(true, true)
+checked expr   : and<Boolean, Boolean>(true, true)
+optimized expr : true
+output type    : Boolean
+output domain  : {TRUE}
+output         : true
+
+
+ast            : false AND false
+raw expr       : and(false, false)
+checked expr   : and<Boolean, Boolean>(false, false)
+optimized expr : false
+output type    : Boolean
+output domain  : {FALSE}
+output         : false
+
+
+ast            : false AND null
+raw expr       : and(false, NULL)
+checked expr   : and<Boolean NULL, Boolean NULL>(CAST(false AS Boolean NULL), CAST(NULL AS Boolean NULL))
 optimized expr : NULL
-output type    : NULL
-output domain  : {NULL}
+output type    : Boolean NULL
+output domain  : Unknown
 output         : NULL
```

Comment:
Now the false and null will return false. Why we modify this?

Reply:
See the description in this pr.